ansible_collections/arista/avd/plugins/plugin_utils/eos_designs_facts.py

```diff
@@ -1687,9 +1693,34 @@ def overlay(self):
         # Set overlay.evpn_vxlan and overlay.evpn_mpls to differentiate between VXLAN and MPLS use cases.
         evpn_vxlan = evpn and self.evpn_encapsulation == "vxlan"
         evpn_mpls = evpn and self.evpn_encapsulation == "mpls"
+        # Set ipvpn_gateway to trigger ipvpn interworking configuration.
+        ipvpn_gateway = evpn and get(self._switch_data_combined, "ipvpn_gateway.enabled", default=False)
         # Set overlay.vpn_ipv4 and vpn_ipv6 to enable IP-VPN configuration on the node.
-        vpn_ipv4 = self.bgp and self.overlay_routing_protocol == "ibgp" and "vpn-ipv4" in self.overlay_address_families
-        vpn_ipv6 = self.bgp and self.overlay_routing_protocol == "ibgp" and "vpn-ipv6" in self.overlay_address_families
+        vpn_ipv4 = self.bgp and (
```

Comment:
The amount of logic justifies separate `_overlay_<flag>` methods, so you just have the last `return` in this method and put the individual logic parts in their own. Use docstring for the helper methods to describe the logic for each area. Thanks

Reply:
I can fix that in a later refactor PR, so we don't need to block this.

Reply:
Added TODO comment in the code.